File: archive/tools-teacher/app/src/main/java/com/instructure/teacheraid/activities/LoginActivity.java
/*
 * Copyright (C) 2016 - present  Instructure, Inc.
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, version 3 of the License.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.instructure.teacheraid.activities;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;

import com.instructure.canvasapi.model.User;
import com.instructure.canvasapi.utilities.ErrorDelegate;
import com.instructure.loginapi.login.URLSignIn;
import com.instructure.loginapi.login.util.Const;
import com.instructure.teacheraid.R;
import com.instructure.teacheraid.util.ApplicationManager;
import com.instructure.teacheraid.util.CanvasErrorDelegate;


public class LoginActivity extends URLSignIn {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //we don't want any help icons because we don't have any zendesk or web resources to link to
        getCanvasHelpIconView().setVisibility(View.GONE);
        getCanvasHelpView().setVisibility(View.GONE);
    }

    @Override
    public ErrorDelegate getErrorDelegate() {
        return new CanvasErrorDelegate();
    }

    @Override
    public void startNextActivity() {
        startActivity(BaseActivity.createIntent(LoginActivity.this));

        finish();
    }

    @Override
    public void startNextActivity(Uri passedURI) {
        startActivity(BaseActivity.createIntent(LoginActivity.this, passedURI));
        finish();
    }

    @Override
    public void initializeLoggingForUser(boolean isSimonFraser, User signedInUser) {



    }

    @Override
    public void startCrashlytics() {
    }

    @Override
    public void startHelpShift() {

    }

    @Override
    public void startGoogleAnalytics() {
       // EasyTracker easyTracker = EasyTracker.getInstance(LoginActivity.this);
       // easyTracker.send(MapBuilder.createEvent("app_flow", "User Signed In", "WelcomeScreen", null).build());
    }

    @Override
    public void showHelpShiftSupport() { }

    @Override
    public void trackAppFlow(Activity activity) {
        //Analytics.trackAppFlow(LoginActivity.this);
    }

    @Override
    public void displayMessage(String message, int messageType) {
        if(!TextUtils.isEmpty(message)){
           Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public int getRootLayout() {
        return R.layout.teacher_tools_url_sign_in;
    }

    @Override
    public String getPrefsFileName() {
        return ApplicationManager.PREF_FILE_NAME;
    }

    @Override
    public String getPrefsPreviousDomainKey() {
        return ApplicationManager.PREF_NAME_PREVIOUS_DOMAINS;
    }

    @Override
    public String getPrefsOtherSignedInUsersKey() {
        return ApplicationManager.OTHER_SIGNED_IN_USERS_PREF_NAME;
    }

    @Override
    public String getPrefsMultiUserKey() {
        return ApplicationManager.MULTI_SIGN_IN_PREF_NAME;
    }

    @Override
    public boolean shouldShowHelpButton() {
        return false;
    }

    ///////////////////////////////////////////////////////////////////////////
    // Intents
    ///////////////////////////////////////////////////////////////////////////

    public static Intent createIntent(Context context) {
        Intent intent = new Intent(context, LoginActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
        return intent;
    }

    public static Intent createIntent(Context context, Uri passedURI) {
        Intent intent = new Intent(context, LoginActivity.class);
        intent.putExtra(Const.PASSED_URI, passedURI);
        intent.setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
        return intent;
    }


    public static Intent createIntent(Context context, boolean showMessage, String message) {
        Intent intent = new Intent(context, LoginActivity.class);
        intent.putExtra(Const.SHOW_MESSAGE, showMessage);
        intent.putExtra(Const.MESSAGE_TO_USER, message);
        intent.setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
        return intent;
    }
}

File: apps/parent/src/main/java/com/instructure/parentapp/activity/LoginActivity.kt
/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, version 3 of the License.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.instructure.parentapp.activity

import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.core.content.ContextCompat
import android.webkit.CookieManager
import com.instructure.canvasapi2.models.User
import com.instructure.canvasapi2.utils.ApiPrefs
import com.instructure.interactions.router.Route
import com.instructure.loginapi.login.activities.BaseLoginInitActivity
import com.instructure.pandautils.utils.Utils
import com.instructure.parentapp.BuildConfig
import com.instructure.parentapp.R

class LoginActivity : BaseLoginInitActivity() {

    override fun launchApplicationMainActivityIntent(): Intent {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            CookieManager.getInstance().flush()
        }

        val intent = Intent(this, SplashActivity::class.java)

        if (getIntent() != null && getIntent().extras != null) {
            intent.putExtras(getIntent().extras)
        }

        return intent
    }

    override fun beginLoginFlowIntent() = LoginLandingPageActivity.createIntent(this)
    override fun themeColor() = ContextCompat.getColor(this, R.color.login_parentAppTheme)
    override fun userAgent() = "androidParent"

    override fun finish() {
        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
        super.finish()
    }

    override fun isTesting(): Boolean {
        return BuildConfig.IS_TESTING
    }

    /**
     * ONLY USE FOR UI TESTING
     * Skips the traditional login process by directly setting the domain, token, and user info.
     */
    fun loginWithToken(token: String, domain: String, user: User) {
        ApiPrefs.token = token
        ApiPrefs.domain = domain
        ApiPrefs.user = user
        ApiPrefs.userAgent = Utils.generateUserAgent(this, userAgent())
        finish()
        val intent = Intent(this, SplashActivity::class.java).apply { intent?.extras?.let { putExtras(it) } }
        startActivity(intent)
    }

    companion object {
        fun createIntent(context: Context): Intent {
            val intent = Intent(context, LoginActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            return intent
        }

        fun createIntent(context: Context, route: Route): Intent {
            val intent = Intent(context, LoginActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            intent.putExtra(Route.ROUTE, route)
            return intent
        }
    }
}
